walkthroughs/week-5-tdd/project/src/main/java/com/google/sps/FindMeetingQuery.java

```diff
@@ -14,10 +14,104 @@
 
 package com.google.sps;
 
+import java.util.ArrayList;
 import java.util.Collection;
+import java.util.Collections;
+import java.util.List;
+import java.util.Set;
 
+/**
+ * Processes a query to find time ranges for a new meeting given the attendees required and other
+ * events of the day.
+ */
 public final class FindMeetingQuery {
+  /**
+   * Finds all the available time slots for a meeting given the time required for the meeting, the
+   * people that need to attend that meeting, and other events scheduled for the day. This
+   * algorithm is O(1) in terms of space complexity, requiring a boolean array of fixed size
+   * (24 * 60). It is O(mn) in time complexity, 'm' being the number of attendees required for the
+   * meeting and 'n' being the events of the day.
+   * @param events all the events scheduled for a given day, with their attendees and time range.
+   * @param request the request for a meeting, with their required attendees and minimum duration.
+   * @return the available time slots during the day where setting the meeting is possible.
+   */
   public Collection<TimeRange> query(Collection<Event> events, MeetingRequest request) {
-    throw new UnsupportedOperationException("TODO: Implement this method.");
+    boolean[] unavailableMinutes = new boolean[24 * 60];
+    Collection<String> meetingAttendees = request.getAttendees();
```

Comment:
Can you note that this is a HashSet? Because otherwise you would be adding more complexity to your computation (because a non-HashSet collection would have to iterate through for the .contains(...) call you make later).

Reply:
Sorry, I don't understand your question. In the class where it's implemented, I can see that it is a HashSet. So should I declare my variable as a HashSet?
![image](https://user-images.githubusercontent.com/45107360/88090604-a84b2b00-cb42-11ea-875d-8e849610fa85.png)

Reply:
That's not necessary, but adding the comment will be good so people looking at your code understand a lookup is O(1) (though actually, you call the lookup on eventAttendees, not meetingAttendees, so make sure it's clear when eventAttendees.contains is called that it is an O(1) computation)

Reply:
It's a HashSet as well. Added it to the comment!